backend/app/rag/indices/vector_search/filters/filter_evaluator.py

```diff
@@ -0,0 +1,277 @@
+                if not FilterEvaluator._validate_array_operator(operator, filter_value):
+                    return False if not sql_mode else None
+            elif operator in FilterEvaluator.TEXT_OPERATORS:
+                if not FilterEvaluator._validate_text_operator(operator, filter_value):
+                    return False if not sql_mode else None
+            elif operator in FilterEvaluator.BASIC_OPERATORS:
+                if not FilterEvaluator._validate_basic_operator(operator, filter_value):
+                    return False if not sql_mode else None
+            elif operator == FilterOperator.IS_EMPTY:
+                # IS_EMPTY doesn't require value validation
+                pass
+            else:
+                logger.error(f"Unsupported operator: {operator}")
+                return False if not sql_mode else None
+
+            # Operator logic processing
+            if operator == FilterOperator.EQ:
+                return meta_column == filter_value if sql_mode else value == filter_value
+            elif operator == FilterOperator.NE:
+                return meta_column != filter_value if sql_mode else value != filter_value
+            elif operator == FilterOperator.GT:
+                return meta_column > filter_value if sql_mode else value > filter_value
+            elif operator == FilterOperator.GTE:
+                return meta_column >= filter_value if sql_mode else value >= filter_value
+            elif operator == FilterOperator.LT:
+                return meta_column < filter_value if sql_mode else value < filter_value
+            elif operator == FilterOperator.LTE:
+                return meta_column <= filter_value if sql_mode else value <= filter_value
+            elif operator == FilterOperator.IN:
+                return meta_column.in_(filter_value) if sql_mode else value in filter_value
+            elif operator == FilterOperator.NIN:
+                return ~meta_column.in_(filter_value) if sql_mode else value not in filter_value
+            elif operator == FilterOperator.CONTAINS:
+                return meta_column.contains(filter_value) if sql_mode else any(item in value for item in filter_value)
+            elif operator == FilterOperator.IS_EMPTY:
+                if sql_mode:
+                    return or_(
+                        meta_column.is_(None),
+                        meta_column == "",
+                        meta_column == "[]"
+                    )
+                return value is None or value == "" or (isinstance(value, list) and len(value) == 0)
+            elif operator == FilterOperator.TEXT_MATCH:
+                if sql_mode:
+                    return meta_column.contains(str(filter_value))
+                return str(filter_value) in str(value)
+            elif operator == FilterOperator.TEXT_MATCH_INSENSITIVE:
+                if sql_mode:
+                    return func.lower(meta_column).contains(func.lower(str(filter_value)))
```

Comment:
`meta_column.icontains` is better

https://docs.sqlalchemy.org/en/20/core/sqlelement.html#sqlalchemy.sql.expression.ColumnElement.icontains